irods/test/pam_interactive_test.py

```diff
@@ -0,0 +1,74 @@
+
+import unittest
+import os
+import json
+from irods.client_init import write_pam_interactive_irodsA_file
+from unittest.mock import patch
+from irods.auth import ClientAuthError
+from irods.auth.pam_interactive import (
+    _pam_interactive_ClientAuthState,
+    PERFORM_WAITING,
+    PERFORM_WAITING_PW,
+    PERFORM_AUTHENTICATED,
+)
+from irods.test.helpers import make_session
+
+class PamInteractiveTest(unittest.TestCase):
+
+    def setUp(self):
+        self.sess = None
+        self.env_file_path = os.path.expanduser("~/.irods/irods_environment.json")
+        self.auth_file_path = os.path.expanduser("~/.irods/.irodsA")
+
+        with open(self.env_file_path) as f:
+            env = json.load(f)
+        self.user = env.get("irods_user_name", "alice")
+        self.zone = env.get("irods_zone_name", "tempZone")
+        self.password = "rods"
+
+    def tearDown(self):
+        if self.sess:
+            self.sess.cleanup()
+
+    def test_pam_interactive_login_basic(self):
+        with patch("getpass.getpass", return_value=self.password):
+            self.sess = make_session(test_server_version=False, env_file=self.env_file_path, authentication_scheme="pam_interactive")
+            self.assertTrue(self.sess.server_version)
+            self.assertEqual(self.sess.username, self.user)
+            self.assertEqual(self.sess.zone, self.zone)
+
+    def test_pam_interactive_auth_file_creation(self):
+        with patch("getpass.getpass", return_value=self.password):
+            write_pam_interactive_irodsA_file(env_file=self.env_file_path)
+            self.assertTrue(os.path.exists(self.auth_file_path), ".irodsA file was not created")
+
+            self.sess = make_session(test_server_version=False, env_file=self.env_file_path, authentication_scheme="pam_interactive")
```

Comment:
Is it important to know whether the password was prompted here? I would expect that the creation of the irodsA file above would preclude the password prompt. I noticed that we assert that `mock_getpass` was `called` in `test_forced_interactive_flow` so it seemed to make sense that we would assert that it was not `called` in this test (but maybe not?).

Reply:
This looks like it has been addressed, but I will leave resolving to you just in case I missed something.